jax/_src/scipy/optimize/minimize.py

```diff
@@ -94,39 +91,47 @@ def minimize(
   Returns:
     An :class:`OptimizeResults` object.
   """
+  try:
+    import jaxopt
+  except ImportError as err:
+    raise RuntimeError("jaxopt package must be installed to use jax.scipy.optimize.minimize") from err
+
+  if not isinstance(args, tuple):
+    raise TypeError(f"args {args} must be a tuple")
+
+  def fun_with_args(x):
+    return fun(x, *args)
+
   if options is None:
     options = {}
 
-  if not isinstance(args, tuple):
-    msg = "args argument to jax.scipy.optimize.minimize must be a tuple, got {}"
-    raise TypeError(msg.format(args))
+  if tol is None:
+    if gtol := options.pop('gtol', None) is not None:
```

Comment:
Should be 
`if (gtol := options.pop('gtol', None)) is not None:`
Otherwise gtol is set to True and that explained the poor behavior.

Reply:
Done, thanks!